icu4c/source/i18n/datefmt.cpp

```diff
@@ -279,9 +280,8 @@ UnicodeString&
 DateFormat::format(UDate date, UnicodeString& appendTo, FieldPosition& fieldPosition) const {
     if (fCalendar != nullptr) {
         UErrorCode ec = U_ZERO_ERROR;
-        const auto* calType = fCalendar->getType();
         // Avoid a heap allocation and corresponding free for the common case
-        if (uprv_strcmp(calType, "gregorian") == 0) {
+        if (typeid(GregorianCalendar) == typeid(*fCalendar)) {
```

Comment:
I suggest you flip the order of these comparisons to increase readability by following the ordering used in English: `if (typeid(*fCalendar) == typeid(GregorianCalendar))` reads like _"if this calendar is a Gregorian calendar"_.